android/jni/AndroidVulkanContext.cpp

```diff
@@ -169,22 +169,24 @@ bool AndroidVulkanContext::InitFromRenderThread(ANativeWindow *wnd, int desiredB
 	return success;
 }
 
-void AndroidVulkanContext::Shutdown() {
+void AndroidVulkanContext::ShutdownFromRenderThread() {
 	ILOG("AndroidVulkanContext::Shutdown");
 	draw_->HandleEvent(Draw::Event::LOST_BACKBUFFER, g_Vulkan->GetBackbufferWidth(), g_Vulkan->GetBackbufferHeight());
-	ILOG("Calling NativeShutdownGraphics");
-	NativeShutdownGraphics();
 	delete draw_;
 	draw_ = nullptr;
 	g_Vulkan->WaitUntilQueueIdle();
-	g_Vulkan->DestroyObjects();
+	g_Vulkan->PerformPendingDeletes();
+	g_Vulkan->DestroyObjects();  // Also destroys the surface, a bit asymmetric
+	ILOG("Done with ShutdownFromRenderThread");
+}
+
+void AndroidVulkanContext::Shutdown() {
+	ILOG("Calling NativeShutdownGraphics");
```

Comment:
Hmm, should it actually call NativeShutdownGraphics here?

-[Unknown]

Reply:
No. Should delete the comment though.